synapse/logging/opentracing.py

```diff
@@ -909,8 +909,9 @@ def tag_args(func: Callable[P, R]) -> Callable[P, R]:
     @wraps(func)
     def _tag_args_inner(*args: P.args, **kwargs: P.kwargs) -> R:
         argspec = inspect.getfullargspec(func)
-        for i, arg in enumerate(argspec.args[1:]):
-            set_tag("ARG_" + arg, str(args[i]))  # type: ignore[index]
+        # We use `[1:]` to skip the `self` object reference
```

Comment:
Should we call out this in the docstring? Something like e.g.

```
`func` is assumed to be a method taking a `self` parameter, or a `classmethod` taking a `cls` parameter. In either case, a tag is not created for this parameter.
```

Reply:
Note for the future: it looks like `inspect.signature` automatically skips the self/cls for us:

```python
>>> class C:
...     def foo(): ...
... 
>>> 
>>> C.foo
<function C.foo at 0x7fafdbd97400>
>>> C().foo
<bound method C.foo of <__main__.C object at 0x7fafdbcd5db0>>
>>> inspect.signature(C.foo)
<Signature ()>
>>> sig = inspect.signature(C.foo); sig
<Signature ()>    
>>> sig.parameters
mappingproxy(OrderedDict())

```

Reply:
> ```py
> >>> class C:
> ...     def foo(): ...
> ```

I think that's missing the `self` parameter? When I test on Python 3.8, `inspect.signature` only omits self for bound methods:

```py
>>> class A:
...     def f(self, x): pass
...
>>> inspect.signature(A.f)
<Signature (self, x)>
>>> inspect.signature(A().f)
<Signature (x)>
```

Reply:
Oh bugger, you're completely right.

Reply:
Added the docstring 👍